sylvia-derive/src/check_generics.rs

```diff
@@ -1,26 +1,51 @@
 use syn::visit::Visit;
-use syn::GenericParam;
+use syn::{GenericArgument, GenericParam, Type};
+
+pub trait AsIdent {
```

Comment:
I really struggle with this. How is there a situation when you want to use `GenericParam` and `GenericArgument` exchangeably? What is even more worrying - in the case of `GenericArgument` implementation disregards any non-path argument, and for paths - it ignores most of the path.